### ISSUE TITLE
More tests needed

### ISSUE THREAD

**Issue by WrinklyNinja** [loot/loot#460]
Right now, the `dev` branch sorting tests are failing for some reason, and I get different results when I run the tests on my local virtual machine. To help pinpoint where exactly things are going wrong, more tests need to be written. The recent code refactoring should make this easier. The following classes need to be tested:

- [x] `ConditionGrammar`
- [x] `ConditionalMetadata`
- [x] `File`
- [x] `FormID`
- [x] `Game`
- [x] `GameSettings`
- [x] `GitHelper`
- [x] `Language`
- [x] `LoadOrderHandler`
- [x] `Location`
- [x] `Masterlist`
- [x] `Message`
- [x] `MessageContent`
- [x] `MetadataList`
- [x] `Plugin`
- [x] `PluginDirtyInfo`
- [x] `PluginLoader`
- [x] `PluginMetadata`
- [x] `PluginSorter`
- [x] `Tag`
- [x] `Version`

For classes with YAML I/O helpers, they should also be tested. In addition, the non-OO functions in the following files need to be tested:

- [x] `game.h`
- [x] `game_settings.h`
- [x] `git_helper.h`
- [x] `helpers.h`

**Comment by WrinklyNinja** [comment]
`GitHelper::FixRepoPermissions()` has no test implemented because it involves modifying file permissions, which isn't possible on Travis' newer infrastructure, and it's a bit of a hack anyway, so I'm not too bothered about it.

**Comment by WrinklyNinja** [comment]
I have created a `testing-metadata` repository to hold a minimal metadata file for LOOT to use when testing. This can be used for testing the API and the MetadataList and Masterlist classes, and will be faster than using the actual game repositories.

I've also added skeleton tests that fail for the remaining functions for which tests have not been written.

In order to test the remaining classes, I need to add that minimal metadata file mentioned above, and add some records to the testing plugins, so that the plugin loader and sorter have some data to work with.
